Source/API/CBLReplication.m

```diff
@@ -343,7 +343,7 @@ - (void) updateStatus: (CBLReplicationStatus)status
         return;
     if (status == kCBLReplicationStopped) {
         _started = NO;
-        [_database forgetReplication: self];
+        [self forgetReplication];
```

Comment:
I don't think removing the observer here is needed, because this is invoked by `_bg_updateProgress`, which when the replicator stops calls `_bg_setReplicator:nil`, which removes the observer.